src/SubscriptionModule.sol

```diff
@@ -1,22 +1,191 @@
+import { CirclesLib } from "src/libs/CirclesLib.sol";
+import { Subscription } from "src/libs/Types.sol";
+import { SubscriptionLib } from "src/libs/SubscriptionLib.sol";
 
-contract SubscriptionModule is Module {
-    constructor(address _owner, address _avatar, address _target) {
-        bytes memory initParams = abi.encode(_owner, _avatar, _target);
-        setUp(initParams);
+import { IHubV2 } from "@circles/src/hub/IHub.sol";
+import { TypeDefinitions } from "@circles/src/hub/TypeDefinitions.sol";
+import { Enum } from "@safe-smart-account/contracts/common/Enum.sol";
+import { EnumerableSetLib } from "@solady/src/utils/EnumerableSetLib.sol";
+
+contract SubscriptionModule {
+    /*//////////////////////////////////////////////////////////////
+                               LIBRARIES
+    //////////////////////////////////////////////////////////////*/
+
+    using SubscriptionLib for Subscription;
+
+    using EnumerableSetLib for EnumerableSetLib.Bytes32Set;
+
+    using CirclesLib for TypeDefinitions.FlowEdge[];
+
+    using CirclesLib for TypeDefinitions.Stream;
+
+    /*//////////////////////////////////////////////////////////////
+                            STATE VARIABLES
+    //////////////////////////////////////////////////////////////*/
+
+    string public constant NAME = "Subscription Module";
+    string public constant VERSION = "0.0.1";
+
+    address public constant HUB = 0xc12C1E50ABB450d6205Ea2C3Fa861b3B834d13e8;
+
+    mapping(bytes32 id => address safe) public safeFromId;
+
+    mapping(address safe => mapping(bytes32 id => Subscription subscription)) public subscriptions;
+
+    mapping(address safe => EnumerableSetLib.Bytes32Set) internal ids;
+
+    mapping(address safe => mapping(uint256 space => mapping(uint256 nonce => bool isRevoked))) internal _revokedNonce;
+
+    mapping(address safe => uint256 space) internal _nonceSpace;
+
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    event SubscriptionCreated(bytes32 indexed id, Subscription indexed subscription);
+
+    event Redeemed(bytes32 indexed id, Subscription indexed subscription);
+
+    event NonceRevoked(address indexed owner, uint256 indexed space, uint256 indexed nonce);
+
+    event NonceSpaceRevoked(address indexed owner, uint256 indexed space);
+
+    /*//////////////////////////////////////////////////////////////
+                                 ERRORS
+    //////////////////////////////////////////////////////////////*/
+
+    error ExecutionFailed();
+
+    error InvalidAmount();
+
+    error IdentifierExists();
+
+    error InvalidFrequency();
+
+    error InvalidRecipient();
+
+    error InvalidSubscriber();
+
+    error NonceAlreadyRevoked(address addr, uint256 space, uint256 nonce);
+
+    error NotRedeemable();
+
+    error SingleStreamOnly();
+
+    error SubscriptionCancelled();
```

Comment:
Just a thought. Given the number of errors, I'd say we're could have an IErrors.sol file.